src/Firehose.Web/Infrastructure/NewCombinedFeedSource.cs

```diff
@@ -0,0 +1,228 @@
+using Firehose.Web.Extensions;
+using Microsoft.Extensions.Caching.Memory;
+using Polly;
+using Polly.Caching.Memory;
+using Polly.Wrap;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.ServiceModel.Syndication;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace Firehose.Web.Infrastructure
+{
+    public class NewCombinedFeedSource
+    {
+        private static HttpClient httpClient;
+        private static AsyncPolicyWrap policy;
+
+        public IEnumerable<IAmACommunityMember> Tamarins { get; }
+
+        public NewCombinedFeedSource(IEnumerable<IAmACommunityMember> tamarins)
+        {
+            EnsureHttpClient();
+
+            Tamarins = tamarins;
+
+			if (policy == null)
+			{
+				// cache in memory for an hour
+				var memoryCache = new MemoryCache(new MemoryCacheOptions());
+				var memoryCacheProvider = new MemoryCacheProvider(memoryCache);
+				var cachePolicy = Policy.CacheAsync(memoryCacheProvider, TimeSpan.FromHours(1));
+
+				// retry policy with max 2 retries, delay by x*x^1.2 where x is retry attempt
+				// this will ensure we don't retry too quickly
+				var retryPolicy = Policy.Handle<FeedReadFailedException>()
+					.WaitAndRetryAsync(2, retry => TimeSpan.FromSeconds(retry * Math.Pow(1.2, retry)));
+
+				policy = Policy.WrapAsync(cachePolicy, retryPolicy);
+			}
+        }
+        
+        private void EnsureHttpClient()
+        {
+            if (httpClient == null)
+            {
+                httpClient = new HttpClient();
+                httpClient.DefaultRequestHeaders.UserAgent.Add(
+                    new ProductInfoHeaderValue("PlanetXamarin", $"{GetType().Assembly.GetName().Version}"));
+                httpClient.Timeout = TimeSpan.FromSeconds(10);
+
+                ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11;
```

Comment:
should this be changed to remove tls11 and add tls13?

Reply:
Yeah, I guess that would be the better practice.

Reply:
TSL13 doesn't seem to be available in full framework